cmake/external/gsl.cmake

```diff
@@ -0,0 +1,16 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+include(FetchContent)
+
+FetchContent_Declare(
+    GSL
+    GIT_REPOSITORY https://github.com/microsoft/gsl
```

Comment:
You may use a https url so that we don't need to download all versions of gsl.

Reply:
would GIT_SHALLOW help with that?

Reply:
Then why not just use the URL https://github.com/microsoft/GSL/archive/refs/tags/v4.0.0.zip ? Like what we have in https://github.com/microsoft/onnxruntime/blob/main/cmake/external/abseil-cpp.cmake

Reply:
It's based on what was recommended here:
https://github.com/microsoft/GSL/blob/1e0d0446dddfa019dae17ffff4398bf3b96ad6d4/README.md?plain=1#L205-L211

Also, I hope we can come up with a way to update cgmanifest entries automatically for these dependencies. This seems easier to parse than a URL.

Reply:
Ok, We can discuss it later.